### ISSUE TITLE
False positive when calling object.__new__ inside __new__ with args

### ISSUE THREAD

**Issue by li-dan** [python/mypy#4190]
Consider:

```python
class A:
    def __new__(cls, x: int) -> 'A':
        return object.__new__(cls) # Argument 1 to "__new__" of "object" has incompatible type "Type[A]"; expected "Type[object]"

class B:
    def __new__(cls) -> 'B':
        return object.__new__(cls)
```

The `object.__new__(cls)` call type checks when inside a `__new__` method with a single argument, but not when there are extra arguments.

The code for `A` is taken from `testSuperWithNew`, but because `object.__new__` is unannotated in `fixtures/__new__.pyi`, the type error doesn't show up in the unit test. The discrepancy between the fixture and typeshed was noted in #3111.

**Comment by ilevkivskyi** [comment]
I think https://github.com/python/mypy/issues/1021 is closely related.

**Comment by msullivan** [comment]
Interestingly, the following works:
```from typing import Type

class A:
    def __new__(cls, x: int) -> 'A':
        cls2: Type[object] = cls
        return object.__new__(cls2)
```

**Comment by msullivan** [comment]
I think the fix for this is to make the type of `cls` be `Type[A]`.
